cmd/krew/cmd/system.go

```diff
@@ -50,7 +53,24 @@ This command will be removed without further notice from future versions of krew
 	PreRunE: ensureIndexUpdated,
 }
 
+var indexUpgradeCmd = &cobra.Command{
+	Use:   "index-upgrade",
+	Short: "Perform a migration of the krew index",
+	Long: `Krew became more awesome! To use the new features, you need to run this
+one-time migration, which will enable installing plugins from custom indexes.
+
+This command will be removed without further notice from future versions of krew.
+`,
+	Args: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return indexmigration.Migrate(paths)
+	},
+}
+
 func init() {
+	if _, ok := os.LookupEnv("X_KREW_ENABLE_MULTI_INDEX"); ok {
```

Comment:
Maybe add the migration check behind this feature gate to the PreRunE of root as well?

Reply:
Sure I can add that, I didn't include the check in root but it makes sense to include it in this PR